src/transformers/models/aria/modular_aria.py

```diff
@@ -0,0 +1,1357 @@
+            raise TypeError("grid_pinpoints must be a list of possible resolutions.")
+
+        possible_resolutions = grid_pinpoints
+
+        image_size = get_image_size(image, channel_dim=input_data_format)
+        best_resolution = select_best_resolution(image_size, possible_resolutions)
+        resized_image = self._resize_for_patching(
+            image, best_resolution, resample=resample, input_data_format=input_data_format
+        )
+        padded_image = self._pad_for_patching(resized_image, best_resolution, input_data_format=input_data_format)
+
+        patches = divide_to_patches(padded_image, patch_size=patch_size, input_data_format=input_data_format)
+
+        # make sure that all patches are in the input data format
+        patches = [
+            to_channel_dimension_format(patch, channel_dim=data_format, input_channel_dim=input_data_format)
+            for patch in patches
+        ]
+        return patches
+
+
+class AriaProcessorKwargs(ProcessingKwargs, total=False):
+    _defaults = {
+        "text_kwargs": {
+            "padding": False,
+            "truncation": None,
+            "max_length": None,
+        },
+        "images_kwargs": {
+            "max_image_size": 980,
+            "split_image": False,
+        },
+        "return_tensors": TensorType.PYTORCH,
+    }
+
+
+class AriaProcessor(ProcessorMixin):
+    """
+    AriaProcessor is a processor for the Aria model which wraps the Aria image preprocessor and the LLama slow tokenizer.
+    Args:
+        image_processor(AriaImageProcessor): The AriaImageProcessor to use for image preprocessing.
+        tokenizer(AutoTokenizer): The AutoTokenizer to use for tokenizing the text.
+        patch_size(int): The patch size to use for the image processor.
+        chat_template(str): The chat template to use for the tokenizer.
+        image_token(str): The image token to use for the tokenizer.
+    """
+
+    attributes = ["image_processor", "tokenizer"]
+    valid_kwargs = ["chat_template", "patch_size", "image_token"]
+    image_processor_class = "AutoImageProcessor"
+    tokenizer_class = "AutoTokenizer"
+
+    def __init__(
+        self,
+        image_processor: AriaImageProcessor = None,
+        tokenizer: Union[AutoTokenizer, str] = None,
+        patch_size: int = 490,
+        chat_template: str = None,
+        image_token: str = "<|img|>",
+        size_conversion: Optional[Dict] = None,
+    ):
+        if size_conversion is None:
+            size_conversion = {490: 128, 980: 256}
+        self.size_conversion = size_conversion
+
+        if image_processor is None:
+            self.image_processor = AriaImageProcessor(max_image_size=patch_size)
+        else:
+            self.image_processor = image_processor
+
+        if isinstance(tokenizer, str):
+            self.tokenizer = AutoTokenizer.from_pretrained(tokenizer, trust_remote_code=True, use_fast=False)
+        else:
+            self.tokenizer = tokenizer
+
```

Comment:
image processor and tokenizer should be automatically loaded and defined when we call `super()`, no need to do it manually